packages/bsky/src/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -79,21 +79,47 @@ export const skeleton = async (inputs: {
   if (clearlyBadCursor(params.cursor)) {
     return { actor, filter: params.filter, items: [] }
   }
+
+  const isFirstPageRequest = !params.cursor
+  const hasPinnedPost =
+    isFirstPageRequest && params.includePins && !!actor.profile?.pinnedPost
+
   const res = await ctx.dataplane.getAuthorFeed({
     actorDid: did,
     limit: params.limit,
     cursor: params.cursor,
     feedType: FILTER_TO_FEED_TYPE[params.filter],
   })
+
+  let items = res.items.map((item) => ({
+    post: { uri: item.uri, cid: item.cid || undefined },
+    repost: item.repost
+      ? { uri: item.repost, cid: item.repostCid || undefined }
+      : undefined,
+  }))
+
+  if (hasPinnedPost && actor.profile?.pinnedPost) {
```

Comment:
Nit: we're checking `pinnedPost` twice here. Maybe `s/hasPinnedPost/shouldInsertPinnedPost` is more clear? The comment below here could also move up to clarify what we're doing: if we have a pinned post, we only insert and filter it for the first page.

Reply:
yeah agreed, renamed 👍